sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -170,6 +170,13 @@ class SessionCatalog(
     tableRelationCache.invalidate(key)
   }
 
+  /** Discards any cached table relation plans for the given table identifier. */
+  private[sql] def invalidateCachedTable(name: TableIdentifier): Unit = {
+    val dbName = formatDatabaseName(name.database.getOrElse(currentDb))
+    val tableName = formatTableName(name.table)
+    invalidateCachedTable(QualifiedTableName(dbName, tableName))
```

Comment:
if the `name` refers to a temp view, we shouldn't call `invalidateCachedTable` here which may accidentally invalid cache for a table with the same name as the temp view.

Reply:
This will not cause any correctness issue, right?

ok. Look at `SessionCatalog.refreshTable`:
https://github.com/apache/spark/blob/157b72ac9fa0057d5fd6d7ed52a6c4b22ebd1dfc/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L990-L1006

if `name` is a temp view, the function could invalidate a record in `tableRelationCache` for a table. but nothing bad happens, `tableRelationCache` will be filled in next time.

Reply:
it doesn't matter too much but still better to avoid. If it's already the case before, then we don't need to fix it here.

Reply:
Seems it's not the case before according to https://github.com/apache/spark/pull/31206/files#r561316669 , @MaxGekk can you fix it?

Reply:
ok. I will wrap the invalidation by the check:
```scala
    if (!sessionCatalog.isTemporaryTable(tableIdent)) {
      sessionCatalog.invalidateCachedTable(tableIdent)
    }
```

Reply:
Here is a fix for `SessionCatalog.refreshTable()`: https://github.com/apache/spark/pull/31265